client/extensions/woocommerce/app/promotions/fields/form-field.js

```diff
@@ -16,21 +18,50 @@ const FormField = ( {
 	labelText,
 	explanationText,
 	isRequired,
+	isEnableable,
+	defaultValue,
 	children,
+	value,
+	edit,
 } ) => {
+	const isValueValid = ! ( 'undefined' === typeof value );
+	const showChildren = ( isEnableable ? isValueValid : true );
+
 	const explanation = ( explanationText &&
 		<FormSettingExplanation id={ fieldName + '-description' }>
 			{ explanationText }
 		</FormSettingExplanation>
 	);
 
+	let enableCheckbox = null;
+
+	if ( isEnableable ) {
+		const enableCheckboxChanged =
+			() => edit( fieldName, ( isValueValid ? undefined : ( defaultValue || null ) ) );
+
+		enableCheckbox = (
+			<FormCheckbox
+				checked={ isValueValid }
+				onChange={ enableCheckboxChanged }
+			/>
+		);
+	}
+
+	const childrenClassNames = classNames(
+		'fields__fieldset-children',
+		{ 'fields__fieldset-children-enableable': enableCheckbox }
```

Comment:
Optional. Replacing enableCheckbox with isEnableable would seem less odd to me. :oldmanyellsatcloud:

Reply:
In this case, it's actually using the enable checkbox as a binary check to see if it should add the class.